insights/client/connection.py

```diff
@@ -676,6 +676,28 @@ def unregister(self):
             logger.error("Could not unregister this system")
             return False
 
+    def unregister(self):
+        """
+        Unregister this system from the insights service
+        """
+        if self.config.legacy_upload:
+            return self._legacy_unregister()
+
+        results = self._fetch_system_by_machine_id()
+        msg_name = determine_hostname(self.config.display_name)
+        try:
+            logger.debug("Unregistering %s", msg_name)
+            url = self.api_url + "/inventory/v1/hosts/" + results[0]['id']
+            net_logger.info("DELETE %s", url)
+            self.session.delete(url)
```

Comment:
Should at least check the HTTP status here to make sure it was successful.

Reply:
Maybe raise requests.ConnectionError so it just jumps into the except block if its a non-2xx code, or whatever this call is supposed to return.